client/rest-high-level/src/main/java/org/elasticsearch/client/eql/EqlSearchRequest.java

```diff
@@ -57,6 +58,7 @@
     static final String KEY_SIZE = "size";
     static final String KEY_FETCH_SIZE = "fetch_size";
     static final String KEY_QUERY = "query";
+    static final String KEY_RESULT_POSITION = "result_position";
```

Comment:
This is about oldest or newest sequences first.
Maybe change this in a true/false option and call it `newest_first` with default `true`. Overriding the default means `false`, implicitly newest first == false == newest last == oldest first.

Reply:
This one is tricky. +1 to Andrei's idea of converting this to a Boolean with a param name of `most_recent_first` or `newest_first`. That would address most of my concerns.

If we decide to keep this as an enum, here are some suggestions in order of preference:

* `head_or_tail`
* `sort_orientation` with `earliest` and `most_recent` (or `latest`) values
* `recency` with `earliest` and `most_recent` (or `latest`) values
* `result_end`

I don't think there is a common phrase that encapsulates both the `head`/`tail` concepts. `end` may work but is probably a bit too vague to be useful here. The `head`/`tail` values make the purpose of the enum param much clearer. As these values are unlikely to change, I'd advocate for including them in the param name if we keep this as an enum.

Reply:
Most recent / orientation suggests a different order. When one does tail the results are still ordered ASC but one gets the last X results not the first X.
I ended up with head and tail since we already have those in the languages as pipes and it's actually what happens if you don't specify them. We could use something like `default_pipe` but then that one is too cryptic...

Reply:
> I ended up with head and tail since we already have those in the languages as pipes and it's actually what happens if you don't specify them.

That makes sense. Rather than forcing another term, it may be simpler and more intuitive to lean into that language:

* `head_or_tail` with `head` and `tail` enum values
* A `head` true/false option with a default of `false` OR a `tail` true/false option with a default of `true`

It seems like the other options are too vague.

Reply:
Forcing "head" or "tail" in the name of the parameter will implicitly assume the user knows about the purpose of head or tail and I like it. So, theoretically, there shouldn't be a concern about some implied ordering the user would assume. In Elasticsearch there are some settings that have technical terms in them and, unless one reads the documentation and knows something more about Elasticsearch, that setting will not make sense. And I think the events selection window (from start or end of the time stream) can be categorized as a more advanced feature that needs a bit more knowledge about EQL. One has to understand which events are selected and that the order in which these events are returned is always ascending.

Also, since we assume the user knows about "head" and "tail" we can push this a bit more and add "default" and "pipe" to the name of the parameter and make it boolean: `"default_head_pipe": true` to be more explicit. Using `default_pipe` is too abstract and it doesn't suggest a limited list of pipes to choose from, which `default_head_pipe` refers explicitly to `head` and it's negation (a value of `false`) implies `tail`.

Reply:
Booleans are more akin to on/off switches. `default_head_pipe` false simply means there's no default head pipe not that there is a tail one instead.